src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -266,30 +266,38 @@ protected virtual void Initialize()
                     File.WriteAllText(hostConfigFilePath, "{}");
                 }
 
-                if (ScriptConfig.HostConfig.IsDevelopment || InDebugMode)
+                var hostConfig = ScriptConfig.HostConfig;
+                if (hostConfig.IsDevelopment || InDebugMode)
                 {
                     // If we're in debug/development mode, use optimal debug settings
-                    ScriptConfig.HostConfig.UseDevelopmentSettings();
+                    hostConfig.UseDevelopmentSettings();
+                }
+
+                // Ensure we always have an ILoggerFactory,
+                // regardless of whether AppInsights is registered or not
+                if (hostConfig.LoggerFactory == null)
```

Comment:
Hoisting this out of ConfigureLoggerFactory so it can be done earlier before any config validation, etc. This ensures that a default is set.